eng/cake/dotnet.cake

```diff
@@ -111,13 +111,18 @@ Task("dotnet-templates")
             { "CustomBeforeMicrosoftCSharpTargets", MakeAbsolute(File("./src/Templates/TemplateTestExtraTargets.targets")).FullPath },
         };
 
-        var frameworks = new [] {
-            "net6.0-android",
-            "net6.0-ios",
-            "net6.0-maccatalyst",
+        var templates = new Dictionary<string, Action<DirectoryPath>> {
+            { "maui:maui", null },
+            { "mauiblazor:maui-blazor", null },
+            { "mauilib:mauilib", null },
+            { "mauicorelib:mauilib", dir => {
+                CleanDirectories(dir.Combine("Platforms").FullPath);
+                ReplaceTextInFiles($"{dir}/*.csproj", "UseMaui", "UseMauiCore");
+                ReplaceTextInFiles($"{dir}/*.csproj", "SingleProject", "EnablePreviewMsixTooling");
```

Comment:
We have to do a few things for maui core as that is not a single project system yet. Maybe it should be? But that is a Q for another day.